cesium/internal/unary/iterator.go

```diff
@@ -24,6 +24,7 @@ type Iterator struct {
 	alamos.Instrumentation
 	IteratorConfig
 	Channel  core.Channel
+	db       *DB
```

Comment:
I think it might be a good idea not to pass in the entire DB here since we're only modifying the counter. What about passing in an anonymous function that we just call on release?